### ISSUE TITLE
JIT: Investigate relationship of basic block's `bbCodeOffs` + `bbCodeOffsEnd` and its statements' `DebugInfo`

### ISSUE THREAD

**Issue by TIHan** [dotnet/runtime#100079]
When looking at issue https://github.com/dotnet/runtime/issues/99543 , a fair amount of time was spent trying to understand how the relationship of a basic blocks' IL offsets correspond to the statements they contain.

What is not clear is what should happen to the basic blocks' IL offsets if the statements they contain are modified/transformed (even at Debug time).

**Comment by JulieLeeMSFT** [comment]
Good point. I think we are not consistent in maintaining IL offset when merging/spliting basic blocks.

**Comment by amanasifkhalid** [comment]
This isn't critical to .NET 10 -- pushing this out
